cocos/3d/CCMesh.cpp

```diff
@@ -613,8 +613,8 @@ void Mesh::setLightUniforms(Pass* pass, Scene* scene, const Vec4& color, unsigne
         if (hasAmbient)
         {
             ambient.x /= 255.f; ambient.y /= 255.f; ambient.z /= 255.f;
+            glProgramState->setUniformVec4("u_color", Vec4(color.x * ambient.x, color.y * ambient.y, color.z * ambient.z, color.w));
```

Comment:
Yeah, No matter we have ambient light or not, we must set the uniform `u_color`.
 I would suggest we do it like this.
```cpp
if(hasAmbients)
{
 glProgramState->setUniformVec4("u_color", a, b, c);
}
else
{
 glProgramState->setUniformVec4("u_color", d, e, f);
}
```

In this code fragment, it is easy to confuse others, because we do not setUniform here, but set it in another function.